R/theme-elements.R

```diff
@@ -144,6 +146,22 @@ element_text <- function(family = NULL, face = NULL, colour = NULL,
   )
 }
 
+#' @export
+#' @rdname element
+is_element <- function(x, type = "any") {
+  switch(
+    type %||% "any",
+    any   = inherits(x, "element"),
+    rect  = inherits(x, "element_rect"),
+    line  = inherits(x, "element_line"),
+    text  = inherits(x, "element_text"),
+    blank = inherits(x, "element_blank"),
+    # TODO: ideally we accept more elements from extensions. We need to
+    # consider how this will work with S7 classes, where ggplot2 doesn't know
+    # about the extension's class objects.
+    FALSE
+  )
+}
```

Comment:
Thomas, I'd like to draw your eye to this bit with regard to the comment here

Reply:
I don't think we _need_ to support anything but our own classes here tbh